opennms-web-api/src/main/java/org/opennms/web/svclayer/support/DefaultGraphResultsService.java

```diff
@@ -154,6 +165,25 @@ public GraphResults findResults(ResourceId[] resourceIds, String[] reports, Stri
                 }
             }
         }
+        // GraphSelected case where all resources are fetched from generatedId
+        if (!Strings.isNullOrEmpty(generatedId) && m_jsonStore != null) {
```

Comment:
If the jsonStore is null, this will just get silently skipped. Probably better to let line 170 throw an NPE so we know why it failed.
```suggestion
        if (!Strings.isNullOrEmpty(generatedId)) {
```

Reply:
It is sanity check. We don't want to throw NPE in the application

Reply:
Wouldn't we rather throw an NPE than silently fail? How will the user know what went wrong here?

There is lots of cases where we explicitly throw an NPE when sanity checking, thats why we use `Objects.requireNonNull()`.

You could change the setter so that it does the `nonNull` check and then remove this null check from the conditional.

Reply:
Actually, the `afterPropertiesSet()` already does the equivalent. So you could simply remove the null check from the conditional.

Reply:
If you look at the changes completely, it's already there. https://github.com/OpenNMS/opennms/pull/2679/files#diff-9c9a5247b8b6fcade3daaae74fe929c9R301

Reply:
> Actually, the `afterPropertiesSet()` already does the equivalent. So you could simply remove the null check from the conditional.
Having null check is okay. I don't want to go through another iteration.

Reply:
I think I missed it from the force push.

Reply:
> I think I missed it from the force push.

It existed from start of this PR

Reply:
> > I think I missed it from the force push.
> 
> It existed from start of this PR

I was referring to your removal of the null check in the conditional. It was there in previous commit and removed after force push. So I didn't see the it change in a new commit which is why I missed it.